cmake/checks/cpu_rvv.cpp

```diff
@@ -6,19 +6,16 @@
 #endif
 
 #if defined CV_RVV
-#if defined(__riscv_v_intrinsic) &&  __riscv_v_intrinsic>10999
-#define vreinterpret_v_u64m1_u8m1 __riscv_vreinterpret_v_u64m1_u8m1
-#define vle64_v_u64m1 __riscv_vle64_v_u64m1
-#define vle32_v_f32m1 __riscv_vle32_v_f32m1
-#define vfmv_f_s_f32m1_f32 __riscv_vfmv_f_s_f32m1_f32
+#if !defined(__riscv_v_intrinsic) ||  (__riscv_v_intrinsic < 12000)
```

Comment:
Let's make this file compatible with T-Head toolchain too. It uses old intrinsics, doesn't define `__riscv_v_intrinsics`, but have `__riscv_v` and `__THEAD_VERSION__` macros. Perhaps even simple `#include <riscv_vector.h>` would be enough to check that it works.

Otherwise our CPU detection mechanism will not find RVV and would disable RVV 0.7.1 HAL: https://github.com/opencv/opencv/blob/53a5b85d9f6e2a58066f966d901c0e93ea1ebcf0/CMakeLists.txt#L1040-L1046

Reply:
Ok, do you mean ...
```cpp
#include <stdio.h>

#if defined(__riscv)
#  include <riscv_vector.h>
#  define CV_RVV 1
#endif

#if defined CV_RVV && !defined __THEAD_VERSION__
#if !defined(__riscv_v_intrinsic) ||  (__riscv_v_intrinsic < 12000)
#error "RISC-V Vector Intrinsic v0.12 or higher is required."
#endif
int test()
{
    const float src[] = { 0.0f, 0.0f, 0.0f, 0.0f };
    uint64_t ptr[2] = {0x0908060504020100, 0xFFFFFFFF0E0D0C0A};
    vuint8m1_t a = __riscv_vreinterpret_v_u64m1_u8m1(__riscv_vle64_v_u64m1(ptr, 2));
    vfloat32m1_t val = __riscv_vle32_v_f32m1((const float*)(src), 4);
    return (int)__riscv_vfmv_f_s_f32m1_f32(val);
}
#elif defined CV_RVV && defined __THEAD_VERSION__
int test()
{
    const float src[] = { 0.0f, 0.0f, 0.0f, 0.0f };
    uint64_t ptr[2] = {0x0908060504020100, 0xFFFFFFFF0E0D0C0A};
    vuint8m1_t a = vreinterpret_v_u64m1_u8m1(vle64_v_u64m1(ptr, 2));
    vfloat32m1_t val = vle32_v_f32m1((const float*)(src), 4);
    return (int)vfmv_f_s_f32m1_f32(val);
}
#else
#error "RISC-V vector extension(RVV) is not supported"
#endif

int main()
{
  printf("%d\n", test());
  return 0;
}
```

Reply:
Yes, it should work.

Perhaps we could also simplify the file a bit to reduce macros depth:
```.cpp
#if !defined(__riscv) || !defined(__riscv_v)
#error "Not RISC-V or no RVV ..."
#endif

#if !defined(__THEAD_VERSION__) && defined(__riscv_v_intrinsic) && __riscv_v_intrinsic < 12000
#error "Wrong intrinsics ..."
#endif

#include <riscv_vector.h>

#ifdef (__THEAD_VERSION__)
int test()
{
    // ... old version
}
#else
{
     // ... new version
}
#endif

int main() {...}
```

Reply:
Updated